File: problems/inorder_tree_traversal/InorderTraversal.java
package problems.inorder_tree_traversal;

import java.util.List;
import java.util.ArrayList;

class TreeNode {
    int val;
    TreeNode left;
    TreeNode right;

    TreeNode() {
    }

    TreeNode(int val) {
        this.val = val;
    }

    TreeNode(int val, TreeNode left, TreeNode right) {
        this.val = val;
        this.left = left;
        this.right = right;
    }
}

class InorderTraversal {
    public List<Integer> inorderTraversal(TreeNode root) {
        // inorder : left , root , right
        List<Integer> order = new ArrayList<>();
        helper(root, order);
        return order;

    }

    private void helper(TreeNode root, List<Integer> order) {
        if (root == null)
            return;
        helper(root.left, order);
        order.add(root.val);
        helper(root.right, order);
    }
}

File: problems/inorder_tree_traversal/inorder_traversal.py
from typing import Optional, List


# Definition for a binary tree node.
class TreeNode:
    def __init__(self, val=0, left=None, right=None):
        self.val = val
        self.left = left
        self.right = right


class Solution:
    def inorderTraversal(self, root: Optional[TreeNode]) -> List[int]:
        order = []
        self.helper(root, order)
        return order

    def helper(self, root: Optional[TreeNode], order):
        if not root:
            return
        self.helper(root.left, order)
        order.append(root.val)
        self.helper(root.right, order)
